### PR TITLE
add ``FilePath`` and ``DirectoryPath`` types #10

### PR DESCRIPTION
**Pull request by Gr1N**
#10

### CODE REVIEW

**Comment by Gr1N** [comment]
@samuelcolvin please review

File: tests/test_types.py [hunk 1]
```diff
@@ -919,6 +919,76 @@ class Model(BaseModel):
...
+        },
+    ]),
+    ('tests/test_types.py', Path('tests/test_types.py')),
+))
+def test_file_path_validation(value, result):
+    class Model(BaseModel):
+        foo: FilePath
+
+    if not isinstance(result, Path):
+        with pytest.raises(ValidationError) as exc_info:
```
Comment (samuelcolvin) [inline tests/test_types.py]: Please separate the two cases into different years.

Comment (Gr1N) [reply tests/test_types.py]: Sorry can't understand what you want (different years?), can you explain more detailed?

Comment (samuelcolvin) [reply tests/test_types.py]: Sorry, I'm on holiday - typo from my phone. 

Separate into different tests.

Comment (Gr1N) [reply tests/test_types.py]: Updated, please review

**Pull request merged**